src/plots.jl

```diff
@@ -0,0 +1,151 @@
+    end
+end
+function _coordvecs(::PolygonTrait, geom)
+    ring = first(getgeom(geom)) # currently doesn't plot holes
+    points = getpoint(ring)
+    if is3d(geom)
+        return getcoord.(points, 1), getcoord.(points, 2), getcoord.(points, 3)
+    else
+        return getcoord.(points, 1), getcoord.(points, 2)
+    end
+end
+function _coordvecs(::MultiPolygonTrait, geom)
+    function loop!(vecs, geom)
+        i1 = 1
+        for ring in getring(geom)
+            i2 = i1 + npoint(ring) - 1
+            range = i1:i2
+            vvecs = map(v -> view(v, range), vecs)
+            _geom2coordvecs!(vvecs..., ring)
+            map(v -> v[i2+1] = NaN, vecs)
+            i1 = i2 + 2
+        end
+        return vecs
+    end
+    n = npoint(geom) + nring(geom)
+    if is3d(geom)
+        vecs = ntuple(_ -> Array{Float64}(undef, n), 3)
+        return loop!(vecs, geom)
+    else
+        vecs = ntuple(_ -> Array{Float64}(undef, n), 2)
+        return loop!(vecs, geom)
+    end
+end
+
+
+_coordvec(n) = Array{Float64}(undef, n)
+
+function _geom2coordvecs!(xs, ys, geom)
+    for (i, p) in enumerate(getpoint(geom))
+        xs[i] = x(p)
+        ys[i] = y(p)
+    end
+    return xs, ys
+end
+function _geom2coordvecs!(xs, ys, zs, geom)
+    for (i, p) in enumerate(getpoint(geom))
+        xs[i] = x(p)
+        ys[i] = y(p)
+        zs[i] = z(p)
+    end
+    return xs, ys, zs
+end
+
+function expr_enable(typ)
+    quote
+        # We recreate the apply_recipe functions manually here
+        # as nesting the @recipe macro doesn't work.
+        function RecipesBase.apply_recipe(plotattributes::Base.AbstractDict{Base.Symbol,Base.Any}, geom::$typ)
+            @nospecialize
+            series_list = RecipesBase.RecipeData[]
+            RecipesBase.is_explicit(plotattributes, :label) || (plotattributes[:label] = :none)
+            Base.push!(series_list, RecipesBase.RecipeData(plotattributes, (GeoInterface.trait(geom), geom)))
+            return series_list
+        end
+        function RecipesBase.apply_recipe(plotattributes::Base.AbstractDict{Base.Symbol,Base.Any}, geom::Base.AbstractVector{<:Base.Union{Base.Missing,<:($typ)}})
+            @nospecialize
+            series_list = RecipesBase.RecipeData[]
+            RecipesBase.is_explicit(plotattributes, :label) || (plotattributes[:label] = :none)
+            for g in Base.skipmissing(geom)
+                Base.push!(series_list, RecipesBase.RecipeData(plotattributes, (GeoInterface.trait(g), g)))
+            end
+            return series_list
+        end
+    end
+end
+
+"""
+     GeoInterfaceRecipes.@enable(GeometryType)
+
+Macro to add plot recipes to a geometry type.
+
+# Usage
+
+```julia
+struct MyGeometry 
+...
+end
+# overload GeoInterface for MyGeometry
+...
+
+# Enable Plots.jl plotting
+GeoInterfaceRecipes.@enable_geo_plots MyGeometry
+```
+"""
+macro enable(typ)
+    esc(expr_enable(typ))
+end
+
+# Compat
+macro enable_geo_plots(typ)
+    esc(expr_enable(typ))
+end
```

Comment:
If I understand correctly, this is no longer available as a "name" to import from GeoInterface / some other package.  Do you want to keep the package GeoInterfaceRecipes and just load that code path as the extension?

Reply:
The macro is exported by GeoInterface now, so it should be usable. I've answered your other question in the general comments.